apps/web/lib/app-providers.tsx

```diff
@@ -48,6 +48,12 @@ type AppPropsWithChildren = AppProps & {
   children: ReactNode;
 };
 
+const getEmbedNamespace = (query: Record<string, string>) => {
+  // Mostly embed query param should be available on server. Use that there.
+  // Use the most reliable detection on client
+  return typeof window !== "undefined" ? window.getEmbedNamespace() : query.embed;
```

Comment:
It avoids possible flash of different theme by getting the correct storage key from backend.
It wasn't possible to do reliably earlier because embed query param was getting lost. After recent fixes, embed query param is reliably forwarded throughh booking page navigations